### ISSUE TITLE
Committing happens to a new branch with the name suffixed `-1`?

### ISSUE THREAD

**Issue by OJFord** [gitbutlerapp/gitbutler#4124]
### Version

0.12.5

### Operating System

Mac OSX

### Distribution Method

dmg

### Describe the issue

When you commit to one of the branches on your virtual branch `branch` & push it seems to create a new branch `origin/branch-1` and the `origin/branch` (nor the local ref) is not updated?

### How to reproduce

Have a virtual branch, make a change, commit it to one of the branches, push that branch

### Expected behavior

Update the existing branch

**Comment by Byron** [comment]
Thanks for reporting!

Following the instructions, I wasn't able to reproduce the issue though. Naming the virtual branch `branch`, and pushing one new commit did just push to `refs/heads/branch` on the remote, a branch that was tracked under `refs/remotes/mine/branch` like I would expect (the remote is called `mine`).

Is there anything else that might be required for a reproduction? Maybe `refs/remotes/mine/branch` already existed when the virtual branch was pushed? Thanks for the clarification.

*I could imagine that `remote/branch` already existed and GitButler created a suffixed branch as to not overwrite the existing tracking branch. That mapping can be changed in the context menu of the virtual branch.

<img width="357" alt="Screenshot 2024-06-19 at 22 06 23" src="https://github.com/gitbutlerapp/gitbutler/assets/63622/ed469e4a-be0f-4b07-92dd-dcbb6b24dde5">

**Comment by OJFord** [comment]
Yes it did already exist, I was trying out GitButler for the first time on an existing project (probably quite a common scenario for newcomers?).

It pushed to the suffixed remote branch, creating it, and left the local branch set to that upstream where previously it had been the unsuffixed one.

I thought I'd enabled allow it to force-push anyway, but either way I think this is confusing behaviour and unlikely to be what you wanted - I would have expected that if it couldn't force push (or lease expired or something) that it would just error and let me work out what I want to do.

**Comment by davidolrik** [comment]
I too experienced this when trying to push to my `main` branch.

![gitbutler-push-to-main](https://github.com/gitbutlerapp/gitbutler/assets/1990/a9e3cd8c-d421-4f3f-9a4a-8f099420e03a)

After clicking push I ended up with a `main-1` branch on the remote.

What I really wanted was just to push a single commit to `main` without a PR.

I clicked "Set remote branch name", and changed it from `main-1` to `main` and pushed again with success.

When this happens, maybe refuse to push before confirmation and then just push to the named branch?
This should be ok to do, because ff there is a mismatch in history the push will be rejected by the server.

**Comment by mtsgrd** [comment]
Hi @OJFord  @davidolrik, we have been primarily focused on supporting a workflow where you push feature branches and merge them into e.g. `main`. That said, you guys aren't the first people to ask about support for pushing directly to master.

If you don't mind me asking, what are your main drivers for wanting to use GitButler outside the scope of concurrent feature branches?

**Comment by OJFord** [comment]
@mtsgrd Mine _was_ a feature branch, called fix-python-reqs or something like that, not master or main.

**Comment by davidolrik** [comment]
> @mtsgrd Mine _was_ a feature branch, called fix-python-reqs or something like that, not master or main.

I think the `-1` but affects all pre-existing branches, mine just happened to be `main`.
Was that also the case for you?

**Comment by OJFord** [comment]
Yes, I mentioned that above: https://github.com/gitbutlerapp/gitbutler/issues/4124#issuecomment-2180358337

As I said there I think that's probably really common for people using GitButler for the first time, not only because it's more likely to have an existing than a brand new project, but because that's where you're going to have the problem that makes the GB solution appealing - at work or otherwise collaborating with many people, on an established project working on multiple things (or with bugs that need a workaround) at once.

**Comment by krlvi** [comment]
Thanks for reporting this, and sorry we left it hanging. This was fixed in https://github.com/gitbutlerapp/gitbutler/pull/5401: when GitButler applies an existing branch, it now preserves that branch name instead of changing it to a suffixed name like `branch-1`.

The fix shipped in `0.13.9`. I’m closing this as resolved, but please open a fresh issue if you still see it on a current release.
